backend/endpoints/threads.py

```diff
@@ -1,3 +1,131 @@
 
 router = APIRouter()
+
+@router.put(
+		"/org/{org_id}/rooms/{room_id}/messages/{message_id}/edithreads",
+		response_model=ResponseModel,
+		status_code=status.HTTP_200_OK,
+		responses={
+				401: {"description": "You are not authorized to edit this message"},
+				404: {"description": "Message not found"},
+				424: {"description": "Message not edited"},
+		},
+)
+async def update_threads_message(
+		org_id: str,
+		room_id: str,
+		message_id: str,
+		request: MessageRequest,
+		background_tasks: BackgroundTasks,
+):
+		"""Updates a message sent in a room.
+
+		Edits an existing message document in the messages database collection while
+		publishing to all members of the room in the background.
+
+		Args:
+				org_id: A unique identifier of the organization.
+				room_id: A unique identifier of the room.
+				message_id: A unique identifier of the message that is being edited.
+				request: A pydantic schema that defines the message request parameters.
+				background_tasks: A background task for publishing to all
+													members of the room.
+
+		Returns:
+				A dict containing data about the message that was edited.
+
+						{
+								"_id": "61c3aa9478fb01b18fac1465",
+								"created_at": "2021-12-22 22:38:33.075643",
+								"edited": true,
+								"emojis": [
+								{
+										"count": 1,
+										"emoji": "👹",
+										"name": "frown",
+										"reactedUsersId": [
+										"619ba4671a5f54782939d385"
+										]
+								}
+								],
+								"files": [],
+								"org_id": "619ba4671a5f54782939d384",
+								"richUiData": {
+								"blocks": [
+										{
+										"data": {},
+										"depth": 0,
+										"entityRanges": [],
+										"inlineStyleRanges": [],
+										"key": "eljik",
+										"text": "HI, I'm mark.. new here",
+										"type": "unstyled"
+										}
+								],
+								"entityMap": {}
+								},
+								"room_id": "619e28c31a5f54782939d59a",
+								"saved_by": [],
+								"sender_id": "619ba4671a5f54782939d385",
+								"text": "string",
+								"threads": []
+						}
+
+		Raises:
+				HTTPException [401]: You are not authorized to edit this message.
+				HTTPException [404]: Message not found.
+				HTTPException [424]: Message not edited.
+		"""
+
+		# payload = Message(**request.dict(), org_id=org_id, room_id=room_id)
+		payload = request.dict()
+
+		loadedMessage = await get_message(org_id, room_id, message_id)
+
+		loadedMessageThread = loadedMessage['threads']
+
+		thread_to_add = {"threads": [payload, *loadedMessageThread]}
+
+		if not loadedMessage:
+				raise HTTPException(
+						status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
+				)
+		
+		# if message["sender_id"] != payload["sender_id"]:
+		#     raise HTTPException(
+		#         status_code=status.HTTP_401_UNAUTHORIZED,
+		#         detail="You are not authorized to edit this message",
+		#     )
+
+		added_thread_message = await edit_message_threads(org_id, message_id, thread_to_add)
```

Comment:
The endpoint is `update_threads_message` what is this line doing?

Reply:
my endpoint is creating a thread under a specified message like the way slack does, you can decided to reply someone message using thread so that is why am using put endpoint because am trying to embed a new message inside an old message which will form a thread